extensions/sql-bindings/src/extension.ts

```diff
@@ -16,22 +17,74 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
 	// register the add sql binding command
 	context.subscriptions.push(vscode.commands.registerCommand('sqlBindings.addSqlBinding', async (uri: vscode.Uri | undefined) => { return launchAddSqlBindingQuickpick(uri); }));
 	// Generate Azure Function command
-	context.subscriptions.push(vscode.commands.registerCommand('sqlBindings.createAzureFunction', async (node: ITreeNodeInfo) => {
-		let connectionInfo = node.connectionInfo;
-		// set the database containing the selected table so it can be used
-		// for the initial catalog property of the connection string
-		let newNode: ITreeNodeInfo = node;
-		while (newNode) {
-			if (newNode.nodeType === 'Database') {
-				connectionInfo.database = newNode.metadata.name;
-				break;
-			} else {
-				newNode = newNode.parentNode;
+	context.subscriptions.push(vscode.commands.registerCommand('sqlBindings.createAzureFunction', async (node?: ITreeNodeInfo) => {
+		let selectedBinding: (vscode.QuickPickItem & { type: BindingType; }) | undefined;
+		let connectionInfo: IConnectionInfo | undefined;
+		let connectionURI: string;
+		let listDatabases: string[] | undefined;
+		let objectName: string | undefined;
+		if (!node) {
+			// if user selects command in command palette we prompt user for information
+			try {
+				// Ask binding type for promptObjectName
+				selectedBinding = await promptForBindingType();
+
+				if (!selectedBinding) {
+					return;
+				}
+				// prompt user for connection profile to get connection info
+				connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+				if (!connectionInfo) {
+					void vscode.window.showWarningMessage(constants.needConnection);
+					return;
+				}
+				connectionURI = await vscodeMssqlApi.connect(connectionInfo);
+				// list databases based on connection profile selected
+				listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
+				const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
+					canPickMany: false,
+					title: constants.selectDatabase,
+					ignoreFocusOut: true
+				}));
+
+				if (!selectedDatabase) {
+					// User cancelled
+					return;
+				}
+				connectionInfo.database = selectedDatabase;
+
+				// prompt user for object name to create function from
+				objectName = await promptForObjectName(selectedBinding.type);
+
+			} catch (e) {
+				void vscode.window.showErrorMessage(getErrorMessage(e));
 			}
+		} else {
+			connectionInfo = node.connectionInfo;
+			// set the database containing the selected table so it can be used
+			// for the initial catalog property of the connection string
+			let newNode: ITreeNodeInfo = node;
+			while (newNode) {
+				if (newNode.nodeType === 'Database') {
+					connectionInfo.database = newNode.metadata.name;
+					break;
+				} else {
+					newNode = newNode.parentNode;
+				}
+			}
+			objectName = generateQuotedFullName(node.metadata.schema, node.metadata.name);
+		}
+		if (!connectionInfo) {
```

Comment:
Yes, since `Argument of type 'IConnectionInfo | undefined' is not assignable to parameter of type 'IConnectionInfo'.
  Type 'undefined' is not assignable to type 'IConnectionInfo'.`.  I thought this as well, but it still complains if I remove this.